### DISCUSSION TITLE
SetConfig

### DISCUSSION THREAD

**Discussion by dmedore** [BrianHumlicek/J2534-Sharp#19] [General]
Hello, First off, thanks for this library which has helped reduce my workload substantially!

I am running into an issue with trying to turn 'LOOP_BACK' on for my channel.  

![image](https://user-images.githubusercontent.com/89533718/130815880-004c20e4-87b5-4289-b8d3-f5d1603c96fe.png)

Things work ok if I skip this SetConfig line, I just don't see messages  that I Tx...

Do you see anything that I am missing/doing incorrectly that is causing the memory issue?  I am using version 1.0.0-CI00031 of the NuGet Package.

Thanks

**Comment by BrianHumlicek** [comment]
What you're doing appears to be ok.  I have looked at the code behind this, and it appears to be solid, and hasn't seen any functional changes in a long time.  I suspect this is happening inside your 2534 driver, but I don't know for certain.  If you could show more of the call stack, that might help.  Also, if you have a different 2534 device you could test with, that might also help.

Brian

**Comment by dmedore** [comment]
I'm not too familiar with J2534 so I wasn't sure if I needed to call a different method prior to the SetConfig.  

I've tried with Vector and PEAK tools PassThru libraries, both being in the PassThruSupport.04.04 registry hive.

I observe the same result with either hardware.  I also observe similar result if I try to use GetConfig

![image](https://user-images.githubusercontent.com/89533718/130887101-db1ac999-0090-4bdf-8932-4b50f5e7be48.png)

![image](https://user-images.githubusercontent.com/89533718/130887351-448aae33-b9e0-4a4a-a15c-c7570e5defd7.png)


Thanks again!

Dan

**Reply by BrianHumlicek** [reply]
Well, thats concerning.  I'll see if I can test it here duplicate it.

**Comment by BrianHumlicek** [comment]
Doing a little google-fu, there are some credible suggestions that you might have mixed build platforms (x86, AnyCPU, x64)

https://stackoverflow.com/questions/4074585/attempted-to-read-or-write-protected-memory-this-is-often-an-indication-that-ot

Check and see if this looks like it might apply to you.

Brian
